packages/dialog/src/lit/renderer-directives.d.ts

```diff
@@ -0,0 +1,88 @@
+/**
+ * @license
+ * Copyright (c) 2017 - 2022 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+/* eslint-disable max-classes-per-file */
+import { TemplateResult } from 'lit';
+import { DirectiveResult } from 'lit/directive.js';
+import { LitRendererDirective } from '@vaadin/lit-renderer';
+import { Dialog } from '../vaadin-dialog.js';
+
+export type DialogLitRenderer = (dialog: Dialog) => TemplateResult;
+
+declare abstract class AbstractDialogRendererDirective extends LitRendererDirective<Dialog, DialogLitRenderer> {
+  /**
+   * A property to that the renderer callback will be assigned.
+   */
+  abstract rendererProperty: string;
+
+  /**
+   * Adds the renderer callback to the dialog.
+   */
+  addRenderer(): void;
+
+  /**
+   * Runs the renderer callback on the dialog.
+   */
+  runRenderer(): void;
+
+  /**
+   * Removes the renderer callback from the dialog.
+   */
+  removeRenderer(): void;
+}
+
+export class DialogRendererDirective extends AbstractDialogRendererDirective {
+  rendererProperty: string;
+}
+
+export class DialogHeaderRendererDirective extends AbstractDialogRendererDirective {
+  rendererProperty: string;
+}
+
+export class DialogFooterRendererDirective extends AbstractDialogRendererDirective {
+  rendererProperty: string;
+}
```

Comment:
```suggestion
export class DialogRendererDirective extends AbstractDialogRendererDirective {
  rendererProperty: 'renderer';
}

export class DialogHeaderRendererDirective extends AbstractDialogRendererDirective {
  rendererProperty: 'headerRenderer';
}

export class DialogFooterRendererDirective extends AbstractDialogRendererDirective {
  rendererProperty: 'footerRenderer';
}
```

Reply:
Good point. I tried it, but it didn't work with `abstract rendererProperty: string`, it should be of literal string type to make it work. Let me change it.

Reply:
Done.